src/coprocessor/dag/expr/arithmetic.rs

```diff
@@ -197,11 +197,121 @@ impl FnCall {
             None => Ok(None),
         }
     }
+
+    pub fn intdivide_int(&self, ctx: &mut EvalContext, row: &[Datum]) -> Result<Option<i64>> {
+        let lhs = try_opt!(self.children[0].eval_int(ctx, row));
+        let rhs = try_opt!(self.children[1].eval_int(ctx, row));
+
+        if rhs == 0 {
+            return Ok(None);
+        }
+
+        let lus = mysql::has_unsigned_flag(self.children[0].get_tp().get_flag());
+        let rus = mysql::has_unsigned_flag(self.children[1].get_tp().get_flag());
+
+        let res = match (lus, rus) {
+            (true, true) => Some(((lhs as u64) / (rhs as u64)) as i64),
+            (false, false) => if lhs == i64::MIN && rhs == -1 {
```

Comment:
Although many of these code are only occurred once, I suggest to extract overflow behaviors into a new module, exactly like what we did in TiDB, to reduce the effort of maintain in future and to make it easier to compare differences between TiDB's implementation and TiKV's implementation.